src/lib.rs

```diff
@@ -1084,3 +1084,84 @@ mod benches {
         });
     }
 }
+
+#[cfg(all(test, feature = "fuzztarget"))]
+mod test_fuzz {
+    use super::*;
+    use std::str::FromStr;
+    use std::panic::{self, UnwindSafe};
+    use std::sync::MutexGuard;
+
+
+    pub fn lock_crypto_var() -> MutexGuard<'static, ()> {
+        use std::sync::{Mutex, Once, ONCE_INIT};
+
+        static mut CRYPTO_VAR_LOCK: Option<Mutex<()>> = None;
+        static INIT_CONTEXT: Once = ONCE_INIT;
+        
+        INIT_CONTEXT.call_once(|| unsafe {
+            CRYPTO_VAR_LOCK = Some(Default::default());
+        });
+        unsafe { CRYPTO_VAR_LOCK.as_ref().unwrap().lock().unwrap() }
+    }
+
+
+
+    pub fn crashed_for_fuzz<F: FnOnce() -> R + UnwindSafe, R>(f: F) -> bool {
+        let description = "Tried fuzzing without setting the UNSAFE_CRYPTO_FUZZING variable";
```

Comment:
Maybe reduce this string just to "UNSAFE_CRYPTO_FUZZING", that's enough for a test.